test/e2e/node/runtimeclass.go

```diff
@@ -0,0 +1,141 @@
+		handler := framework.GetRuntimeClassHandler("test-handler")
+		scheduling := &v1beta1.Scheduling{
+			NodeSelector: map[string]string{
+				"foo": "conflict",
+			},
+		}
+
+		runtimeClass := newRuntimeClass(f.Namespace.Name, "conflict-runtimeclass", handler)
+		runtimeClass.Scheduling = scheduling
+		rc, err := f.ClientSet.NodeV1beta1().RuntimeClasses().Create(runtimeClass)
+		framework.ExpectNoError(err, "failed to create RuntimeClass resource")
+
+		pod := newRuntimeClassPod(rc.GetName())
+		pod.Spec.NodeSelector = map[string]string{
+			"foo": "bar",
+		}
+		_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
+		framework.ExpectError(err)
+	})
+
+	ginkgo.It("should run a Pod requesting a RuntimeClass with scheduling [NodeFeature:RuntimeHandler] ", func() {
+		handler := framework.GetRuntimeClassHandler("test-handler")
+		nodeName := scheduling.GetNodeThatCanRunPod(f)
+		nodeSelector := map[string]string{
+			"foo":  "bar",
+			"fizz": "buzz",
+		}
+		tolerations := []v1.Toleration{
+			{
+				Key:      "foo",
+				Operator: v1.TolerationOpEqual,
+				Value:    "bar",
+				Effect:   v1.TaintEffectNoSchedule,
+			},
+		}
+		scheduling := &v1beta1.Scheduling{
+			NodeSelector: nodeSelector,
+			Tolerations:  tolerations,
+		}
+
+		ginkgo.By("Trying to apply a label on the found node.")
+		for key, value := range nodeSelector {
+			framework.AddOrUpdateLabelOnNode(f.ClientSet, nodeName, key, value)
+			framework.ExpectNodeHasLabel(f.ClientSet, nodeName, key, value)
+			defer framework.RemoveLabelOffNode(f.ClientSet, nodeName, key)
+		}
+
+		ginkgo.By("Trying to apply taint on the found node.")
+		taint := v1.Taint{
+			Key:    "foo",
+			Value:  "bar",
+			Effect: v1.TaintEffectNoSchedule,
+		}
+		framework.AddOrUpdateTaintOnNode(f.ClientSet, nodeName, taint)
+		defer framework.RemoveTaintOffNode(f.ClientSet, nodeName, taint)
+
+		ginkgo.By("Trying to create runtimeclass and pod")
+		runtimeClass := newRuntimeClass(f.Namespace.Name, "non-conflict-runtimeclass", handler)
+		runtimeClass.Scheduling = scheduling
+		rc, err := f.ClientSet.NodeV1beta1().RuntimeClasses().Create(runtimeClass)
+		framework.ExpectNoError(err, "failed to create RuntimeClass resource")
+
+		pod := newRuntimeClassPod(rc.GetName())
+		pod.Spec.NodeSelector = map[string]string{
+			"foo": "bar",
+		}
+		pod = f.PodClient().Create(pod)
+
+		framework.ExpectNoError(e2epod.WaitForPodSuccessInNamespace(f.ClientSet, pod.Name, f.Namespace.Name))
+		framework.ExpectEqual(nodeSelector, pod.Spec.NodeSelector)
+		framework.ExpectEqual(tolerations, pod.Spec.Tolerations)
+
+		// check that pod got scheduled on specified node.
+		labelPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(pod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(labelPod.Spec.NodeName, nodeName)
+	})
+})
+
+// newRuntimeClass returns a test runtime class.
+func newRuntimeClass(namespace, name, handler string) *v1beta1.RuntimeClass {
```

Comment:
nit: all the tests here use the preconfigured runtime handler, so you can omit the handler arg and just inline that function call here.